bentoml/_internal/configuration/containers.py

```diff
@@ -271,26 +265,54 @@ def __init__(
 
             config_merger.merge(self.config, override_config)
 
-            global_runner_cfg = {k: self.config["runners"][k] for k in RUNNER_CFG_KEYS}
-            for key in self.config["runners"]:
-                if key not in RUNNER_CFG_KEYS:
-                    runner_cfg = self.config["runners"][key]
+        if override_config_values is not None:
+            logger.info(
+                "Applying user config override from ENV VAR: %s", override_config_values
+            )
+            lines = split_with_quotes(
+                override_config_values,
+                sep=r"\s+",
+                quote='"',
+                use_regex=True,
+            )
+            override_config_map = {
+                k: yaml.load(v)
+                for k, v in [
+                    split_with_quotes(line, sep="=", quote='"') for line in lines
+                ]
+            }
+            try:
+                override_config = unflatten(override_config_map)  # type: ignore
+            except ValueError as e:
+                raise BentoMLConfigException(
+                    f'Failed to parse config options from the env var: {e}. \n *** Note: You can use " to quote the key if it contains special characters. ***'
+                ) from None
+            config_merger.merge(self.config, override_config)
 
-                    # key is a runner name
-                    if runner_cfg.get("resources") == "system":
-                        runner_cfg["resources"] = system_resources()
+        if override_config_file is not None or override_config_values is not None:
+            self._finish_config()
 
-                    self.config["runners"][key] = config_merger.merge(
-                        deepcopy(global_runner_cfg), runner_cfg
-                    )
+        if validate_schema:
+            try:
+                SCHEMA.validate(self.config)
+            except SchemaError as e:
+                raise BentoMLConfigException(
+                    "Default configuration 'default_configuration.yml' does not"
+                    " conform to the required schema."
+                ) from e
 
-            if validate_schema:
-                try:
-                    SCHEMA.validate(self.config)
-                except SchemaError as e:
-                    raise BentoMLConfigException(
-                        "Invalid configuration file was given."
-                    ) from e
+    def _finish_config(self):
```

Comment:
lets just call this api `finalize`?
```suggestion
    def finalize(self):
```